components/base/wheeled/wheeled_base.go

```diff
@@ -130,64 +130,53 @@ func (wb *wheeledBase) Reconfigure(ctx context.Context, deps resource.Dependenci
 
 	wb.spinSlipFactor = newConf.SpinSlipFactor
 
-	// Check if wb.left is different from newConf.Left before changing wb.left
-	if len(wb.left) != len(newConf.Left) {
-		// Resetting the left motor list
-		wb.left = make([]motor.Motor, 0)
-
-		for _, name := range newConf.Left {
-			m, err := motor.FromDependencies(deps, name)
-			if err != nil {
-				return errors.Wrapf(err, "no left motor named (%s)", name)
+	updateMotors := func(curr []motor.Motor, fromConfig []string) ([]motor.Motor, error) {
```

Comment:
This needs to take in a string telling you which motor array it is. If you name a right motor "1" but put it in the left config you're losing user-facing information on why the motor is not there, and add that string to the error message array